db/db_impl.cc

```diff
@@ -2300,7 +2298,8 @@ Status DestroyDB(const std::string& dbname, const Options& options) {
     // Delete log files in the WAL dir
     for (const auto& file : walDirFiles) {
       if (ParseFileName(file, &number, &type) && type == kLogFile) {
-        Status del = env->DeleteFile(LogFileName(soptions.wal_dir, number));
+        Status del = DeleteDBFile(
+            &soptions, LogFileName(soptions.wal_dir, number), 0, type);
```

Comment:
I'm not sure about using `path_id == 0` when `wal_dir` is set. Feels to me we should treat files that aren't in `db_paths[0]` directory the same. It might be required for correctness, too, as `wal_dir` might not be on the same filesystem as `trash_dir`.